sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/ctl/AsyncCtlWorkload.java

```diff
@@ -0,0 +1,350 @@
+
+        reporter.start(configuration.getPrintingInterval(), TimeUnit.SECONDS);
+        long startTime = System.currentTimeMillis();
+
+        AtomicLong count = new AtomicLong(0);
+        long i;
+        int writeRange = readPct + writePct;
+        for (i = 0; BenchmarkHelper.shouldContinue(startTime, i, configuration); i++) {
+            int index = (int) i % 100;
+            if (index < readPct) {
+                BenchmarkRequestSubscriber<Object> readSubscriber = new BenchmarkRequestSubscriber<>(readSuccessMeter,
+                    readFailureMeter, concurrencyControlSemaphore, count);
+                readSubscriber.context = readLatency.time();
+                performWorkload(readSubscriber, OperationType.Read, i);
+            } else if (index < writeRange) {
+                BenchmarkRequestSubscriber<Object> writeSubscriber = new BenchmarkRequestSubscriber<>(writeSuccessMeter,
+                    writeFailureMeter, concurrencyControlSemaphore, count);
+                writeSubscriber.context = writeLatency.time();
+                performWorkload(writeSubscriber, OperationType.Create, i);
+
+            } else {
+                BenchmarkRequestSubscriber<Object> querySubscriber = new BenchmarkRequestSubscriber<>(querySuccessMeter,
+                    queryFailureMeter, concurrencyControlSemaphore, count);
+                querySubscriber.context = queryLatency.time();
+                performWorkload(querySubscriber, OperationType.Query, i);
+            }
+        }
+
+        synchronized (count) {
+            while (count.get() < i) {
+                count.wait();
+            }
+        }
+
+        long endTime = System.currentTimeMillis();
+        logger.info("[{}] operations performed in [{}] seconds.",
+            configuration.getNumberOfOperations(), (int) ((endTime - startTime) / 1000));
+
+        reporter.report();
+        reporter.close();
+    }
+
+    private void parsedReadWriteQueryPct(String readWriteQueryPct) {
+        String[] readWriteQueryPctList = readWriteQueryPct.split(",");
+        if (readWriteQueryPctList.length == 3) {
+            try {
+                if (Integer.valueOf(readWriteQueryPctList[0]) + Integer.valueOf(readWriteQueryPctList[1]) + Integer.valueOf(readWriteQueryPctList[2]) == 100) {
+                    readPct = Integer.valueOf(readWriteQueryPctList[0]);
+                    writePct = Integer.valueOf(readWriteQueryPctList[1]);
+                    queryPct = Integer.valueOf(readWriteQueryPctList[2]);
+                } else {
+                    logger.warn(PERCENT_PARSING_ERROR, readPct,
+                        writePct, queryPct);
+                }
+            } catch (NumberFormatException ex) {
+                logger.warn(PERCENT_PARSING_ERROR, readPct,
+                    writePct, queryPct);
+            }
+        } else {
+            logger.warn(PERCENT_PARSING_ERROR, readPct, writePct,
+                queryPct);
+        }
```

Comment:
if the input config is not valid, shouldn't we log error and terminate? 
otherwise invalid config may go unnoticed.

Reply:
Yes i was printing warning and using default, but throwing exception would be more clear. Done